src/libpriv/rpmostree-passwd-util.c

```diff
@@ -213,6 +228,152 @@ compare_group_ents (gconstpointer a, gconstpointer b)
+                                                                 g_free, sysuser_ent_free);
+
+  for (int counter=0; counter < group_ents->len; counter++)
+    {
+      struct conv_group_ent *convent = group_ents->pdata[counter];
+      struct sysuser_ent *sysent = g_hash_table_lookup (sysusers_table, convent->name);
+
+      if (!sysent)
+        sysent = g_new (struct sysuser_ent, 1);
+      else
+        {
+          /* For now the simplest thing would be to check if uid:gid is set for "u" */
+          if (g_str_equal (sysent->type, "u"))
+            {
+              g_autofree char *current_gid = g_strdup_printf ("%u", convent->gid);
+              if (g_str_equal (sysent->id, current_gid))
+                {
+                  continue;
+                }
+              char *colon = strchr (sysent->id, ':');
+              if (colon)
+                {
+                  char *stored_gid = colon + 1;
+                  g_print ("current_gid is %s\n", current_gid);
+                  g_print ("stored_gid is %s\n", stored_gid);
+                  /* We skip the insertion if we found gid matches */
+                  if (g_str_equal (stored_gid, current_gid))
+                  {
+                    continue;
+                  }
+                  else
+                  {
+                    *error = g_error_new (G_IO_ERROR, G_IO_ERROR_FAILED, "mismatch between /lib/passwd and /lib/group");
+                    g_print("Catastrophic failure\n");
+                  }
+                }
+
+            }
+        }
+      sysent->id = g_strdup_printf ("%u", convent->gid);
+      sysent->type = g_strdup("g");
+      sysent->name = g_strdup (convent->name);
+      /* Unset the gecos & dir for g entries */
+      sysent->gecos = g_strdup ("-");
+      sysent->dir = g_strdup ("-");
+      char *name = g_strdup (convent->name);
+      g_hash_table_insert (sysusers_table, name, sysent);
+    }
+
+  /* Do the assignment at the end if the sysusers_table was not initialized */
+  if (*out_sysusers_table == NULL)
+    *out_sysusers_table = g_steal_pointer (&sysusers_table);
+
+  return TRUE;
+}
+
+gboolean
+rpmostree_passwd_sysusers2char (GHashTable *sysusers_table,
+                                char      **out_content,
+                                GError    **error)
+{
+  /* Sinc ethe collision is handled during insertion, we just need to
+   * do the conversion directly */
+  GString* sysuser_content = g_string_new (NULL);
+  GLNX_HASH_TABLE_FOREACH_KV (sysusers_table, const char*,  key, struct sysuser_ent*, sysuser_ent)
+    {
+      g_autofree gchar* line_content = g_strjoin(" ", sysuser_ent->type,
+                                                 sysuser_ent->name, sysuser_ent->id,
+                                                 sysuser_ent->gecos, sysuser_ent->dir,
+                                                 NULL);
+      g_string_append_printf(sysuser_content, "%s\n", line_content);
```

Comment:
BTW our code style has space between identifier and paren, so `g_string_append_printf (sysuser_content` and elsewhere.